### DISCUSSION TITLE
Forms: Labels vs. Content

### DISCUSSION THREAD

**Discussion by nofue** [sqlpage/SQLPage#620] [Ideas]
There's one thing with SQLPage I find with most every kind of form builder which doesn't work well in "real life": font sizes / emphasis.

Forms have one single purpose: collect structured data in the easiest/friendliest way possible. 

However, there are two views on forms: first, it's the programmer's view. It's the one which is important through building the form. Fields are usually empty at this stage, and the form "lives and breathes" by the labels. However -- the programmer is the least important person in the life of a form, as he doesn't use it. 

That's the second view on the form: the user's. The user fills in the form and needs all the information gathered being readable at a glance. The user deals with the form often, hence he doesn't need labels. He vaguely knows, in which field which content goes, and he's usually distracted by the labels. 

As a seasoned designer, holding even a form design degree, I suggest to reduce the size of the labels in forms to the same size column headers in tables use. 

**Current look:**
![Bildschirmfoto 2024-09-24 um 07 38 30](https://github.com/user-attachments/assets/5afa383d-86bc-41b0-b40d-f80ebd588901)

**Proposed look:**
![redone](https://github.com/user-attachments/assets/ea7b99be-4801-4403-8ce7-12839d9bcbd0)

Oh, there's three more things:
Why do checkbox areas have to get a blueish background when checked? Isn't the checkmark enough of an indication?
Wouldn't it be nice when the color of the label of an unchecked item would be dimmed to the color of the other labels? Does the checkbox area have to be so much higher than all the other fields?

**Comment by lovasoa** [comment]
Hello, and thank you very much for this detailed recommendation. AlCurrently, we use the default form design of @tabler. 

@codecalm, what do you think about this? Would it be worth changing the defaults in tabler?

**Reply by nofue** [reply]
I can't comment on whether it's worth to deviate from templates -- I just observed that forms are made with a programmer perspective which (most often) get's into the way of users of the forms made. As a designer, I'll always opt for the most user friendly option possible, but as a developer I also know that wishes are free, doing the work isn't.

**Reply by lovasoa** [reply]
For users themselves, it's better to highlight the contents when they are familiar with the form. But what about forms that will be filled only once (or very few times) by each user ? Isn't it better to highlight the labels in this case?

**Reply by nofue** [reply]
Good Question. To solve the puzzle, open a blank form. What do you see, no matter what size the font? Just labels. Only when the form has been filled in the labels will "stand back". Exactly what you intended, right? Even first time users look at the filled in form once before submitting, and then they want to spot typos and such at a glance, which isn't possible with a crowded form. 

The awesome solution would of course be a class like "TRUE as emphasized_labels", but this is even more work, obviously. To me, SQLpage shines by it's lightness and speed and too many options will bloat it as all the other tools got bloated by too many wishes fulfilled …

**Comment by nofue** [comment]
Here's what the empty form will look like with placeholders for date and time in effect:

![redone_empty](https://github.com/user-attachments/assets/13a5a851-5ab8-4281-87d8-a0191d96564e)

**Comment by nofue** [comment]
As things haven't changed with labels in a while, I wonder if there's some CSS available where I can override the label style?

**Comment by lovasoa** [comment]
You can customize your form designs with custom css using the shell component.

## SQL

```sql
select 'shell' as component, '/form_design.css' as css;
```

## `form_design.css`

```css
.form-label {
  color: var(--tblr-secondary);
  font-weight: unset;
  font-variant: small-caps;
  font-size: large;
}
```


<img width="1920" height="878" alt="image" src="https://github.com/user-attachments/assets/844000fd-3260-447c-a25e-5f785543bc4a" />

**Reply by nofue** [reply]
Thank you very much -- I just returned from an exhausting trip, but I'll try this asap.
